openmls/src/credentials/mod.rs

```diff
@@ -120,10 +120,16 @@ impl Credential {
         label: &str,
     ) -> Result<(), CredentialError> {
         match &self.credential {
-            MlsCredentialType::Basic(basic_credential) => basic_credential
-                .public_key
-                .verify_with_label(backend, signature, &SignContent::new(label, payload.into()))
-                .map_err(|_| CredentialError::InvalidSignature),
+            MlsCredentialType::Basic(basic_credential) => {
+                let signature_public_key_enriched = basic_credential
```

Comment:
When we use `SignaturePublicKeyEnriched` in `BasicCredential` we would need to implement TLS (de)serialization for `SignaturePublicKeyEnriched`. I decided against that because these traits should IMO only be implemented for types that are meant to be TLS (de)serialized. So the hope is that this clone will just go away in the future.